llvmlite/ir/values.py

```diff
@@ -14,13 +15,15 @@
 
 _VALID_CHARS = (frozenset(map(ord, string.ascii_letters)) |
                 frozenset(map(ord, string.digits)) |
-                frozenset('._-$'))
+                frozenset(map(ord, ' !#$%&\'()*+,-./:;<=>?@[]^_`{|}~')))
 
 
 def _escape_string(text, _map={}):
     """
     Escape the given bytestring for safe use as a LLVM array constant.
     """
+    if isinstance(text, six.text_type):
```

Comment:
Initially I did `isinstance(text, str)` since otherwise the assert below would fail on Python 3, but that failed on Python 2 because of the superfluous `encode` call...